terraform/cloudwatch.tf.json.template.py

```diff
@@ -173,6 +190,32 @@ def prod_qualified_resource_name(name: str) -> str:
                     }
                     for lambda_ in config.lambda_names()
                 ),
+                *(
+                    {
+                        'aws_cloudwatch_metric_alarm': {
+                            alarm_name: {
+                                'alarm_name': config.qualified_resource_name(alarm_name, suffix='.alarm'),
+                                'namespace': 'AWS/Lambda',
+                                'dimensions': {
+                                    'FunctionName': f'${{aws_lambda_function.{threshold.lambda_name}.function_name}}'
+                                },
+                                'metric_name': threshold.metric.capitalize(),
+                                'comparison_operator': 'GreaterThanThreshold',
+                                'statistic': 'Sum',
+                                'threshold': threshold.threshold,
+                                'period': 5 * 60,
+                                'datapoints_to_alarm': 1,
+                                'evaluation_periods': 1,
+                                'treat_missing_data': 'notBreaching',
+                                'alarm_actions': ['${data.aws_sns_topic.monitoring.arn}'],
+                                'ok_actions': ['${data.aws_sns_topic.monitoring.arn}'],
+                            }
+                        }
+                    }
+                    for lambda_name in config.lambda_names()
+                    for threshold in load_app_module(lambda_name).app.metric_thresholds()
+                    for alarm_name in (alarm_resource_name(threshold),)
```

Comment:
Performing a comprehension over a singleton sequence literal is very peculiar. I'm guessing the intent is to avoid calling the `alarm_resource_name` function twice. I think that because the function is so simple and quick to evaluate, calling it twice is preferable to using this syntax, especially if you accept my suggestion to convert it to a property.